src/radical/pilot/agent/executing/popen.py

```diff
@@ -701,10 +701,15 @@ def _extend_pre_exec(self, td, ranks=None):
 
         if td['gpus_per_rank'] and td['gpu_type'] == rpc.CUDA and ranks:
             # equivalent to the 'physical' value for original `cvd_id_mode`
-            td['pre_exec'].append(
-                {str(rank_id): 'export CUDA_VISIBLE_DEVICES=%s' %
-                               ','.join([str(gm[0]) for gm in rank['gpu_map']])
-                 for rank_id, rank in enumerate(ranks)})
+            rank_id  = 0
+            rank_env = {}
+            for ranks_per_slot in ranks:
```

Comment:
I am confused with naming.  I would think that `ranks_per_slot` is a number, like `2`.  And the `core_map` should contain the mapping of cores to ranks.  I don't think the executor needs to know about the details of the `ranks_per_slot` the schedule had to deal with, does it?

Reply:
should I rename to `slot_ranks` (as in scheduler)?

Reply:
> I don't think the executor needs to know about the details of the ranks_per_slot the schedule had to deal with, does it?

it kinda does, since it sets this env variable per rank